### PR TITLE
Run tests on XSOR_NG

### PR DESCRIPTION
**Pull request by moishce**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -724,38 +723,43 @@ def concurrently_run_function_on_servers(self, function=None, pack_path=None, se
...
+        self.marketplace_buckets = options.marketplace_buckets
 
     @staticmethod
-    def get_xsiam_configuration(xsiam_machine, xsiam_servers_path, xsiam_servers_api_keys_path):
-        xsiam_servers = get_json_file(xsiam_servers_path)
-        conf = xsiam_servers.get(xsiam_machine)
-        xsiam_servers_api_keys = get_json_file(xsiam_servers_api_keys_path)
-        api_key = xsiam_servers_api_keys.get(xsiam_machine)
+    def get_cloud_configuration(cloud_machine, cloud_servers_path, cloud_servers_api_keys_path):
+        logging.info('get cloud configuration')
```
Comment (ShahafBenYakir) [inline Tests/configure_and_test_integration_instances.py:745]:
```suggestion
        logging.info('getting cloud configuration')
```

Comment (ShahafBenYakir) [inline Tests/configure_and_test_integration_instances.py:745]:
```suggestion
        logging.info('getting cloud configuration')
```

File: Tests/configure_and_test_integration_instances.py [hunk 2]
```diff
@@ -1664,8 +1671,8 @@ def create_build_object() -> Build:
     logging.info(f'Build type: {options.build_object_type}')
     if options.build_object_type == XSOAR_BUILD_TYPE:
         return XSOARBuild(options)
-    elif options.build_object_type == XSIAM_BUILD_TYPE:
-        return XSIAMBuild(options)
+    elif options.build_object_type == CLOUD_BUILD_TYPE:
+        return CLOUDBuild(options)
```
Comment (ShahafBenYakir) [inline Tests/configure_and_test_integration_instances.py:1675]:
```suggestion
        return CloudBuild(options)
```

File: .gitlab/ci/on-push.yml [hunk 3]
```diff
@@ -466,12 +458,46 @@ slack-notify-nightly-build:
...
   variables:
     INSTANCE_ROLE: "XSIAM 1.2"
     GCS_QUEUE_FILE: "queue-ga"
-    TEST_MACHINES_LIST: "test-machines-ga"
\ No newline at end of file
+    TEST_MACHINES_LIST: "test-machines-ga"
+    GCS_LOCKS_PATH: "gs://xsoar-ci-artifacts/content-locks-xsoar-ng"
+    CLOUD_SERVERS_FILE: "xsoar_ng_servers_path"
+    CLOUD_API_KEYS: $XSOAR_NG_API_KEYS
+    GCS_MARKET_BUCKET: "${GCS_MARKET_V2_BUCKET}"
```
Comment (ShahafBenYakir) [inline .gitlab/ci/on-push.yml:496]: This should be the regular bucket no?
```suggestion
    GCS_MARKET_BUCKET: "${GCS_MARKET_BUCKET}"
```

**Review by ShahafBenYakir — COMMENTED** [review]
Great work

File: .gitlab/ci/on-push.yml [hunk 4]
```diff
@@ -466,12 +458,46 @@ slack-notify-nightly-build:
...
+    GCS_QUEUE_FILE: "queue-ga"
+    TEST_MACHINES_LIST: "test-machines-ga"
+    GCS_LOCKS_PATH: "gs://xsoar-ci-artifacts/content-locks-xsiam"
+    CLOUD_SERVERS_FILE: "xsiam_servers_path"
+    CLOUD_API_KEYS: $XSIAM_API_KEYS
+    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_MPV2}"
+    GCS_MARKET_BUCKET: "${GCS_MARKET_V2_BUCKET}"
+    ENV_RESULTS_PATH: "${ARTIFACTS_FOLDER_MPV2}/env_results.json"
+    GCS_SOURCE_BUCKET: "$GCS_PRODUCTION_V2_BUCKET"
+    GCS_MACHINES_BUCKET: "marketplace-v2-dist-dev/upload-flow/builds-xsiam"
```
[the remark this answers is not in the document]
Comment (moishce) [reply .gitlab/ci/on-push.yml:413]: This is where the bucket of each machine is saved

File: Tests/configure_and_test_integration_instances.py [hunk 5]
```diff
@@ -1810,7 +1817,7 @@ def main():
         1. Add server config and restart servers (only in xsoar).
         2. Disable all enabled integrations.
         3. Upload all test playbooks that currently in master.
-        4. In XSOAR:Install all existing packs, in XSIAM: install only requested packs.
+        4. In XSOAR:Install all existing packs, in cloud: install only requested packs.
```
[the remark this answers is not in the document]
Comment (moishce) [reply Tests/configure_and_test_integration_instances.py:1845]: The choice of which packs to install happens in the collector
So in the case of NG it’s the same as xsoar

[the remark this answers is not in the document]
Comment (moishce) [reply Tests/configure_and_test_integration_instances.py:1845]: I'll fix the comment

**Review by ShahafBenYakir — APPROVED** [review]

**Comment by moishce** [comment]
@ShahafBenYakir pr is ready just waiting for the build to finish because of a conflict I resolved
and it needs a force merge because of an error in the validate due to deleting files.

`[ERROR]: Tests/scripts/uninstall_packs_and_reset_bucket_xsiam.sh: [BA115] - The file Tests/scripts/uninstall_packs_and_reset_bucket_xsiam.sh cannot be deleted. Please restore the file.

[ERROR]: Tests/scripts/print_xsiam_machine_details.sh: [BA115] - The file Tests/scripts/print_xsiam_machine_details.sh cannot be deleted. Please restore the file.`

**Comment by daryakoval** [comment]
> @ShahafBenYakir pr is ready just waiting for the build to finish because of a conflict I resolved and it needs a force merge because of an error in the validate due to deleting files.
> 
> `[ERROR]: Tests/scripts/uninstall_packs_and_reset_bucket_xsiam.sh: [BA115] - The file Tests/scripts/uninstall_packs_and_reset_bucket_xsiam.sh cannot be deleted. Please restore the file.
> 
> [ERROR]: Tests/scripts/print_xsiam_machine_details.sh: [BA115] - The file Tests/scripts/print_xsiam_machine_details.sh cannot be deleted. Please restore the file.`

@moishce Try to use `git mv` in order to move these files and not delete them when renaming.  https://www.educative.io/answers/how-to-rename-or-move-files-in-git

**Comment by ShahafBenYakir** [comment]
@daryakoval Agree, due to time consideration I will force merge now.

**Pull request merged**